sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -22,6 +23,41 @@
     private Utility() {
     }
 
+    /**
+     * Automatically detect byte buffer's content type.
+     * Given the source: <a href="https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files"/>.
+     *
+     * @param buffer The byte buffer input.
+     *
+     * @return The {@link ContentType} content type.
+     */
+    public static ContentType getContentType(ByteBuffer buffer) {
+        final byte[] bytes = buffer.array();
+        if (bytes.length < 4) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("Invalid input. Expect more than 4 bytes of data"));
+        }
+
+        if (isEqual(bytes[0], 0x25) && isEqual(bytes[1], 0x50) && isEqual(bytes[2], 0x44) && isEqual(bytes[3], 0x46)) {
```

Comment:
nitpick (feel free to ignore)
You could make the function isEqual take in a list of bytes and what they should be, so the code could be more like
`isEqual(bytes.subList(0, 4), [0x25, 0x50, 0x44, 0x46])`